File: Dynamic-Programming/53-M-Maximum-Subarray/max-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int len = nums.length;
        int best=nums[0], sum=nums[0];
        for (int i=1; i<len; i++)
        {
            sum = Math.max(nums[i], sum+nums[i]);
            best = Math.max(best, sum);
        }
        return best;
    }
}

File: Dynamic-Programming/53-M-Maximum-Subarray/max-subarray.c
int maximum (int a, int b)
{
    return a>b?a:b;
}
int maxSubArray(int* nums, int numsSize) {
    int best=nums[0], sum = nums[0];
    for (int k=1; k<numsSize; k++)
    {
        sum = maximum(nums[k], sum+nums[k]);
        best = maximum(best, sum);
    }
    return best;
}